addons/website/models/ir_ui_view.py

```diff
@@ -520,3 +520,11 @@ def _validate_module_views(self, module):
         if module == 'website' or module.startswith('website_') or module.startswith('theme_'):
             generate_primary_snippet_templates(self.env, module)
         return super()._validate_module_views(module)
+
+    @api.model
+    def _adjust_new_page_template(self, html_tree):
+        # data-snippet must be the short general name
+        for section_el in html_tree.xpath("//section[@data-snippet]"):
+            snippet = section_el.attrib['data-snippet']
+            if '_s_' in snippet:
```

Comment:
why?

Reply:
Because the generated page is made of calls such as:
```xml
t-snippet-call="website.new_page_template_about_0_s_text_block"
```
which generates `data-snippet="new_page_template_about_0_s_text_block"` while we want it to be `data-snippet="s_text_block"`.

If your question was about "why `_s_` specifically" - that's because it is the expression I use to split on the next line.

Reply:
I meant that we can't have a random `if '_s_' in snippet:` in the middle of the code without any explanation, it should really be documented with a comment next to it and ideally with an example as you mentioned on top of it